src/main/java/Parser.java

```diff
@@ -0,0 +1,121 @@
+        }
+    }
+
+    private static AddCommand newDeadline(String input) throws DukeException {
+        if (input.equals("deadline") || input.equals("deadline ")) {
+            throw new EmptyDescriptionException("deadline");
+        } else if (input.startsWith("deadline ") && input.length() > 9) {
+            if (!input.contains(" /by ")) {
+                throw new InvalidDateTimeException("deadline");
+            } else {
+                int index = input.indexOf(" /by ");
+                String description = input.substring(9, index);
+                String date = input.substring(index + 5);
+                try {
+                    LocalDate deadlineDate = LocalDate.parse(date);
+                    Deadline newDeadline = new Deadline(description, deadlineDate);
+                    return new AddCommand(newDeadline);
+                } catch (DateTimeParseException e) {
+                    throw new InvalidDateTimeException("deadline");
+                }
+            }
+        } else {
+            throw new DukeException();
+        }
+    }
+
+    private static AddCommand newEvent(String input) throws DukeException {
+        if (input.equals("event") || input.equals("event ")) {
+            throw new EmptyDescriptionException("event");
+        } else if (input.startsWith("event ") && input.length() > 6) {
+            if (!input.contains(" /at ")) {
+                throw new InvalidDateTimeException("event");
+            } else {
+                int index = input.indexOf(" /at ");
+                String description = input.substring(6, index);
+                String date = input.substring(index + 5);
+                try {
+                    LocalDate eventDate = LocalDate.parse(date);
+                    Event newEvent = new Event(description, eventDate);
+                    return new AddCommand(newEvent);
+                } catch (DateTimeParseException e) {
+                    throw new InvalidDateTimeException("event");
+                }
+            }
+        } else {
+            throw new DukeException();
+        }
+    }
+
+    static Command parse(String input) throws DukeException {
+        if (input.equals("bye")) {
+            return new ExitCommand();
+        } else if (input.equals("list") || input.equals("list ")) {
+            return new ListCommand();
+        } else {
+            if (input.startsWith("done")) {
+                return done(input);
+            } else if (input.startsWith("delete")) {
+                return delete(input);
+            } else if (input.startsWith("find")) {
+                return find(input);
+            } else if (input.startsWith("todo")) {
+                return newTodo(input);
+            } else if (input.startsWith("deadline")) {
+                return newDeadline(input);
+            } else if (input.startsWith("event")) {
+                return newEvent(input);
+            } else {
+                throw new DukeException();
```

Comment:
May I suggest using a switch block instead? I find it would make the code more readable since each line would be shorter.
E.g. 
case "deadline":
instead of 
else if (input.startsWith("deadline"))